### PR TITLE
RSVP Fixes

### PR DESCRIPTION
**Pull request by ASankaran**
- [x] Only allow accepted users to rsvp
- [x] Allow user to decline then accept rsvp

### CODE REVIEW

File: api/v1/controllers/RSVPController.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ const config = require('ctx').config();
 
 const router = require('express').Router();
 function _isAuthenticated(req) {
-  return req.auth && (req.user !== undefined);
+  if(!(req.auth && (req.user !== undefined))) {
```
Comment (YashoSharma) [inline api/v1/controllers/RSVPController.js]: Let's combine these booleans

File: api/v1/controllers/RSVPController.js [hunk 2]
```diff
@@ -8,7 +8,10 @@ const config = require('ctx').config();
...
+    return false;
+  }
+  return services.RegistrationService.findAttendeeByUser(req.user).then((attendee) => attendee != null && attendee.get('status') == 'ACCEPTED');
```
Comment (YashoSharma) [inline api/v1/controllers/RSVPController.js]: Let's not use a magic string, there should be a constant in the AttendeeModel

File: api/v1/controllers/RSVPController.js [hunk 3]
```diff
@@ -8,7 +8,10 @@ const config = require('ctx').config();
 
 const router = require('express').Router();
 function _isAuthenticated(req) {
```
Comment (YashoSharma) [inline api/v1/controllers/RSVPController.js]: Update this function name to better represent what it does

File: api/v1/controllers/RSVPController.js [hunk 4]
```diff
@@ -97,7 +100,7 @@ router.post('/attendee', middleware.request(requests.RSVPRequest),
 router.get('/attendee/', middleware.permission(roles.ATTENDEE), fetchRSVPByUser);
 router.get('/attendee/:id(\\d+)', middleware.permission(roles.ORGANIZERS), fetchRSVPById);
 router.put('/attendee/', middleware.request(requests.RSVPRequest),
-  middleware.permission(roles.ATTENDEE), updateRSVPByUser);
+  middleware.permission(roles.ATTENDEE, _isAuthenticated), updateRSVPByUser);
```
Comment (YashoSharma) [inline api/v1/controllers/RSVPController.js]: Make sure there aren't other routers in this controller that use the isAuthenticated method that get effected by the changes.

File: api/v1/controllers/RSVPController.js (hunk 2 above)
Comment (ASankaran) [reply api/v1/controllers/RSVPController.js]: I would need to do Attendee.STATUSES[0] to get the 'ACCEPTED' string, so there's still a magic number.

**Pull request merged**